slave/buildslave/bot.py

```diff
@@ -260,12 +260,20 @@ def __init__(self, name, directory, rotateLength=1000000, defaultMode=None, maxR
 
     def save(self, messages):
         for message in messages:
-            json.dump(message, self)
-            self.write(os.linesep)
+            self.json_serialize(message)
 
         self.flush()
         self.handleFileRotation()
 
+    def json_serialize(self, message):
+        try:
+            json.dump(message, self)
+            self.write(os.linesep)
+        except:
+            log.err(failure.Failure(), "Failed to serialize message to json:")
+            log.msg("message:")
+            log.msg(message)
```

Comment:
@sualitu we can search for "Failed to serialize message to json"  in logz.io since we are collecting twistd.log messages